dll/win32/browseui/shellbrowser.cpp

```diff
@@ -1022,6 +1024,9 @@ HRESULT CShellBrowser::BrowseToPath(IShellFolder *newShellFolder,
     HRESULT                                 hResult;
     //TODO: BOOL                            nohistory = m_BrowserSvcFlags & BSF_NAVNOHISTORY;
 
+    if (m_Destroyed)
```

Comment:
This is the part that looks a bit hackish: I would like to understand (via the stack trace you can get when the crash happens there) what the suite of the calls that leads to here is.

Reply:
Technically just an assert and not a real crash.

```
(The actual assert is in MapListColumnToFolderColumn because the ListView window is gone)
01bcc5d4 7afb0d02 shell32!CDefView::SaveViewState [...\dll\win32\shell32\CDefView.cpp @ 3230]
01bcc5e4 78dd75d2 shell32!_ZN8CDefView13SaveViewStateEv+0x24 [...\dll\win32\shell32\CDefView.cpp @ 3303]
01bcc5fc 78dd7638 browseui!CShellBrowser::SaveViewState+0x2d [...\dll\win32\browseui\shellbrowser.cpp @ 1007]
01bcc87c 78dd727e browseui!CShellBrowser::BrowseToPath+0x50 [...\dll\win32\browseui\shellbrowser.cpp @ 1048]
01bcc8cc 78ddb373 browseui!CShellBrowser::BrowseToPIDL+0xc8 [...\dll\win32\browseui\shellbrowser.cpp @ 904]
01bcc8fc 78dcc1d7 browseui!CShellBrowser::BrowseObject+0x80 [...\dll\win32\browseui\shellbrowser.cpp @ 2366]
01bcc934 78dcc451 browseui!CExplorerBand::UpdateBrowser+0x47 [...\dll\win32\browseui\explorerband.cpp @ 268]
01bcc94c 78dce8d2 browseui!CExplorerBand::OnSelectionChanged+0x2a [...\dll\win32\browseui\explorerband.cpp @ 339]
01bcc994 78dbaa8e browseui!CExplorerBand::OnWinEvent+0x20b [...\dll\win32\browseui\explorerband.cpp @ 1333]
01bcc9c4 78da64ca browseui!CBaseBarSite::OnWinEvent+0x68 [...\dll\win32\browseui\basebarsite.cpp @ 358]
01bcc9f4 78df4485 browseui!CBaseBar::OnNotify+0x65 [...\dll\win32\browseui\shellbars\CBaseBar.cpp @ 509]
01bcca1c 78df1a84 browseui!CBaseBar::ProcessWindowMessage+0x10a [...\dll\win32\browseui\shellbars\CBaseBar.cpp @ 139]
01bcca7c 77a8b656 browseui!CWindowImplBaseT<ATL::CWindow, ATL::CWinTraits<1442840576, 0> >::WindowProc+0x7c [...\sdk\lib\atl\atlwin.h @ 1660]
01bccaac 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bccc1c 77a7fd0e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bccc6c 77a80907 user32!IntCallMessageProc+0x22d [...\win32ss\user\user32\windows\message.c @ 1798]
01bccce4 7be42224 user32!SendMessageW+0x150 [...\win32ss\user\user32\windows\message.c @ 2403]
01bccd94 77a8b656 comctl32!REBAR_WindowProc+0x367 [...\dll\win32\comctl32\rebar.c @ 3728]
01bccdc4 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bccf34 77a7fe9e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bccf64 78df1abb user32!CallWindowProcW+0x51 [...\win32ss\user\user32\windows\message.c @ 1872]
01bccfc4 77a8b656 browseui!CWindowImplBaseT<ATL::CWindow, ATL::CWinTraits<1442840576, 0> >::WindowProc+0xb3 [...\sdk\lib\atl\atlwin.h @ 1616]
01bccff4 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bcd164 77a7fd0e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bcd1b4 77a80907 user32!IntCallMessageProc+0x22d [...\win32ss\user\user32\windows\message.c @ 1798]
01bcd22c 7be72aea user32!SendMessageW+0x150 [...\win32ss\user\user32\windows\message.c @ 2403]
01bcd2dc 7be74afe comctl32!TREEVIEW_SendTreeviewNotify+0xcb [...\dll\win32\comctl32\treeview.c @ 529]
01bcd32c 7be77026 comctl32!TREEVIEW_DoSelectItem+0x19c [...\dll\win32\comctl32\treeview.c @ 4535]
01bcd374 7be777be comctl32!TREEVIEW_SelectItem+0x47 [...\dll\win32\comctl32\treeview.c @ 4580]
01bcd464 77a8b656 comctl32!TREEVIEW_WindowProc+0x642 [...\dll\win32\comctl32\treeview.c @ 5773]
01bcd494 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bcd604 77a7fe9e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bcd634 78df1abb user32!CallWindowProcW+0x51 [...\win32ss\user\user32\windows\message.c @ 1872]
01bcd694 77a8b656 browseui!CWindowImplBaseT<ATL::CWindow, ATL::CWinTraits<1442840576, 0> >::WindowProc+0xb3 [...\sdk\lib\atl\atlwin.h @ 1616]
01bcd6c4 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bcd834 77a7fd0e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bcd884 77a80907 user32!IntCallMessageProc+0x22d [...\win32ss\user\user32\windows\message.c @ 1798]
01bcd8fc 78dcc416 user32!SendMessageW+0x150 [...\win32ss\user\user32\windows\message.c @ 2403]
01bcd924 78dce8ed browseui!CExplorerBand::OnTreeItemDeleted+0xc1 [...\dll\win32\browseui\explorerband.cpp @ 317]
01bcd96c 78dbaa8e browseui!CExplorerBand::OnWinEvent+0x226 [...\dll\win32\browseui\explorerband.cpp @ 1336]
01bcd99c 78da64ca browseui!CBaseBarSite::OnWinEvent+0x68 [...\dll\win32\browseui\basebarsite.cpp @ 358]
01bcd9cc 78df4485 browseui!CBaseBar::OnNotify+0x65 [...\dll\win32\browseui\shellbars\CBaseBar.cpp @ 509]
01bcd9f4 78df1a84 browseui!CBaseBar::ProcessWindowMessage+0x10a [...\dll\win32\browseui\shellbars\CBaseBar.cpp @ 139]
01bcda54 77a8b656 browseui!CWindowImplBaseT<ATL::CWindow, ATL::CWinTraits<1442840576, 0> >::WindowProc+0x7c [...\sdk\lib\atl\atlwin.h @ 1660]
01bcda84 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bcdbf4 77a7fd0e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bcdc44 77a80907 user32!IntCallMessageProc+0x22d [...\win32ss\user\user32\windows\message.c @ 1798]
01bcdcbc 7be42224 user32!SendMessageW+0x150 [...\win32ss\user\user32\windows\message.c @ 2403]
01bcdd6c 77a8b656 comctl32!REBAR_WindowProc+0x367 [...\dll\win32\comctl32\rebar.c @ 3728]
01bcdd9c 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bcdf0c 77a7fe9e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bcdf3c 78df1abb user32!CallWindowProcW+0x51 [...\win32ss\user\user32\windows\message.c @ 1872]
01bcdf9c 77a8b656 browseui!CWindowImplBaseT<ATL::CWindow, ATL::CWinTraits<1442840576, 0> >::WindowProc+0xb3 [...\sdk\lib\atl\atlwin.h @ 1616]
01bcdfcc 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bce13c 77a7fd0e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bce18c 77a80907 user32!IntCallMessageProc+0x22d [...\win32ss\user\user32\windows\message.c @ 1798]
01bce204 7be72aea user32!SendMessageW+0x150 [...\win32ss\user\user32\windows\message.c @ 2403]
01bce2b4 7be72c29 comctl32!TREEVIEW_SendTreeviewNotify+0xcb [...\dll\win32\comctl32\treeview.c @ 529]
01bce2ec 7be72d61 comctl32!TREEVIEW_RemoveItem+0x4d [...\dll\win32\comctl32\treeview.c @ 1496]
01bce314 7be72c05 comctl32!TREEVIEW_RemoveAllChildren+0x1e [...\dll\win32\comctl32\treeview.c @ 1450]
01bce34c 7be72d61 comctl32!TREEVIEW_RemoveItem+0x29 [...\dll\win32\comctl32\treeview.c @ 1493]
01bce374 7be72df1 comctl32!TREEVIEW_RemoveAllChildren+0x1e [...\dll\win32\comctl32\treeview.c @ 1450]
01bce394 7be7680f comctl32!TREEVIEW_RemoveTree+0x11 [...\dll\win32\comctl32\treeview.c @ 1513]
01bce3dc 7be78101 comctl32!TREEVIEW_DeleteItem+0x37 [...\dll\win32\comctl32\treeview.c @ 1577]
01bce4cc 77a8b656 comctl32!TREEVIEW_WindowProc+0xf85 [...\dll\win32\comctl32\treeview.c @ 5690]
01bce4fc 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bce66c 77a7fe9e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bce69c 78df1abb user32!CallWindowProcW+0x51 [...\win32ss\user\user32\windows\message.c @ 1872]
01bce6fc 77a8b656 browseui!CWindowImplBaseT<ATL::CWindow, ATL::CWinTraits<1442840576, 0> >::WindowProc+0xb3 [...\sdk\lib\atl\atlwin.h @ 1616]
01bce72c 77a7f8f4 user32!CALL_EXTERN_WNDPROC+0x1a [...\win32ss\user\user32\windows\wndproc_fixup.S @ 48]
01bce89c 77a7fd0e user32!IntCallWindowProcW+0x1857 [...\win32ss\user\user32\windows\message.c @ 1547]
01bce8ec 77a80907 user32!IntCallMessageProc+0x22d [...\win32ss\user\user32\windows\message.c @ 1798]
01bce964 78dcbc77 user32!SendMessageW+0x150 [...\win32ss\user\user32\windows\message.c @ 2403]
01bce99c 78dce249 browseui!CExplorerBand::DestroyExplorerBand+0x101 [...\dll\win32\browseui\explorerband.cpp @ 170]
01bce9f4 78dbb2ec browseui!CExplorerBand::SetSite+0xf6 [...\sdk\lib\atl\atlwin.h @ 464]
01bcea7c 78dbaf6a browseui!CBaseBarSite::RemoveBand+0xd4 [...\dll\win32\browseui\basebarsite.cpp @ 595]
01bceae4 78ddcf26 browseui!CBaseBarSite::SetDeskBarSite+0x433 [...\dll\win32\browseui\basebarsite.cpp @ 391]
01bceb2c 78de889a browseui!CShellBrowser::OnDestroy+0x283 [...\dll\win32\browseui\shellbrowser.cpp @ 3662]
01bceb5c ....
```